### PR TITLE
XP-485 Document revised state machine

### PR DESCRIPTION
**Pull request by finiteprods**
### References

[XP-240](https://xainag.atlassian.net/browse/XP-240) (in particular [XP-485](https://xainag.atlassian.net/browse/XP-485))

### Summary

This documents the changes introduced in reducing the [Participant state machine](https://github.com/xainag/xain-sdk/pull/47). It updates the content in the Network Architecture documentation with updated diagrams and supporting text, specifically

- update to Coordinator state diagram and supporting text to account for dropouts
- tweak to sequence diagram
- (most significant) update to Participant state diagram and supporting text
- other minor updates

### Are there any open tasks/blockers for the ticket?

None.

---

### Reviewer checklist

*Reviewer agreement:*

* Reviewers assign themselves at the start of the review.
* Reviewers do **not** commit or merge the merge request.
* Reviewers have to check and mark items in the checklist.

**Merge request checklist**

- [x] Conforms to the merge request title naming `XP-XXX <a description in imperative form>`.
- [x] Each commit conforms to the naming convention `XP-XXX <a description in imperative form>`.
- [x] Linked the ticket in the merge request title or the references section.
- [x] Added an informative merge request summary.

**Code checklist**

- [x] Conforms to the branch naming `XP-XXX-<a_small_stub>`.
- [x] Passed scope checks.
- [x] Added or updated tests if needed.
- [x] Added or updated code documentation if needed.
- [x] Conforms to Google docstring style.
- [x] Conforms to XAIN structlog style.

### CODE REVIEW

File: docs/network_architecture.rst [hunk 1]
```diff
@@ -259,31 +259,29 @@ Let's consider the basic lifecycle of state transitions in :math:`C`. Let
...
     graph TB
     A( ) -->|startup| B(STANDBY)
-    B -->|N Ps registered| C(ROUND i)
-    C -->|next round i := i+1| C
+    B -->|N registered| C(ROUND)
+    C -->|dropout| B
+    C -->|next round| C
     C -->|no more rounds| D(FINISHED)
 
-Once :math:`C` starts up, it's in the **STANDBY** state, waiting for incoming
```
Comment (little-dude) [inline docs/network_architecture.rst]: The previous version was slightly clearer imo.

File: docs/network_architecture.rst [hunk 2]
```diff
@@ -375,14 +373,14 @@ Training Round Communication
 
 The communication is summarised in the following sequence diagram.
 In a training round, :math:`C` is in the state :code:`ROUND`. The selected
-participant :math:`P` is in :code:`TRAINING` state. The first message is by
+participant :math:`P` is in the :code:`TRAINING` state (see state diagram later). The first message by
```
Comment (little-dude) [inline docs/network_architecture.rst]: "later" sounds a bit odd to me, maybe

> see state diagram below

File: docs/network_architecture.rst (hunk 1 above)
Comment (finiteprods) [reply docs/network_architecture.rst]: I think "waiting" over-used and there is even a Participant state called that now, so "open" is probably ok!

File: docs/network_architecture.rst (hunk 2 above)
Comment (finiteprods) [reply docs/network_architecture.rst]: added link to section instead

**Pull request merged**